### PR TITLE
Update dependency pipeline with image template and vhd copy

### PR DESCRIPTION
**Pull request by eriqua**
# Change

Closes issue #462 

This change allows the creation of a custom image using Azure VM Image Builder in both the GH and ADO dependency pipelines. 
In particular, the additional jobs are:
- deploying the Azure Image Builder template
- triggering the image build and waiting for its completion
- retrieving the location of the vhd file
- copying the vhd to a known storage account blob container as a dependency for compute/disks and compute/images modules

The custom image creation is optional in the dependency pipeline and disabled by default
image

[.Platform: Dependencies](https://github.com/Azure/ResourceModules/actions/workflows/platform.dependencies.yml)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

## Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (readme)
- [ ] I did format my code

### CODE REVIEW

File: .github/workflows/platform.dependencies.yml [hunk 1]
```diff
@@ -286,6 +291,146 @@ jobs:
...
+            $deploymentOutput = '${{ steps.deploy_imgt.outputs.deploymentOutput }}'
+            $imageTemplateName = (ConvertFrom-Json $deploymentOutput).name
+            $imageTemplateResourceGroup = (ConvertFrom-Json $deploymentOutput).resourceGroupName
+            Write-Verbose "imageTemplateName: $imageTemplateName" -Verbose
+            Write-Verbose "imageTemplateResourceGroup: $imageTemplateResourceGroup" -Verbose
+            Write-Output ('::set-output name={0}::{1}' -f 'imageTemplateName', $imageTemplateName)
+            Write-Output ('::set-output name={0}::{1}' -f 'imageTemplateResourceGroup', $imageTemplateResourceGroup)
+          azPSVersion: 'latest'
+
+  job_deploy_vhd_to_sa:
```
Comment (MrMCake) [inline .github/workflows/platform.dependencies.yml:340]: Would it make sense to merge these steps to the image template deployment? Would avoid the need to pass outputs around

File: utilities/pipelines/dependencies/Microsoft.VirtualMachineImages/imageTemplates/parameters/parameters.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                "sku": "19h2-evd",
+                "version": "latest"
+            }
+        },
+        "customizationSteps": {
+            "value": [
+                {
+                    "type": "WindowsRestart",
+                    "restartTimeout": "30m"
+                }
```
Comment (MrMCake) [inline utilities/pipelines/dependencies/Microsoft.VirtualMachineImages/imageTemplates/parameters/parameters.json:37]: Do we need a step?

File: .github/workflows/platform.dependencies.yml (hunk 1 above)
Comment (eriqua) [reply .github/workflows/platform.dependencies.yml:340]: Outputs would need to be passed even between steps. The main reasons why I've put it in a separate job are: 

1. It is consistent with KV and storage post deployment which are also in a separate job
2. Having them in the same job would mean running a double step for the agent setup. One as a step in the job and one as a step in the deployment action and I found that a bit messy to follow. Anyway we could discuss about it.

File: utilities/pipelines/dependencies/Microsoft.VirtualMachineImages/imageTemplates/parameters/parameters.json (hunk 2 above)
Comment (eriqua) [reply utilities/pipelines/dependencies/Microsoft.VirtualMachineImages/imageTemplates/parameters/parameters.json:37]: `customizationSteps` is a required parameter for the ImageTemplate module. I guess the restart is the one having less impact (it's also the one used by the default module parameters)

**Review by MrMCake — APPROVED** [review]

**Pull request merged**